artifact.md

```diff
@@ -0,0 +1,72 @@
+# OCI Artifact Manifest Specification
+
+The goal of the Artifact Manifest Specification is to define content addressable artifacts in order to store them along side container images in a registry. Like [OCI Images](manifest.md), OCI Artifacts may be referenced by the hash of their manifest. Unlike OCI Images, OCI Artifacts are not meant to be used by any container runtime.
+
+Examples of artifacts that may be stored along with container images are Software Bill of Materials (SBOM), Digital Signatures, Provenance data, Supply Chain Attestations, scan results, and Helm charts.
+
+This section defines the `application/vnd.oci.artifact.manifest.v1+json` [media type](media-types.md).
+For the media type(s) that this is compatible with see the [matrix](media-types.md#compatibility-matrix).
+
+# Artifact Manifest
+
+## *Artifact Manifest* Property Descriptions
+
+- **`mediaType`** *string*
+
+  This property MUST be used and contain the media type `application/vnd.oci.artifact.manifest.v1+json`.
+
```

Comment:
Would it be possible to allow `config` here as `OPTIONAL`?  In my use case, I want to use an OCI artifact for packaging an application and want to use the `config` to provide execution details for the application.

Reply:
You can push an additional blob with a distinct media type. Or you can use the image manifest with a custom config. I don't believe anything we are doing requires an artifact to use the artifact manifest. The existing OCI Artifact guidance using the image manifest is still valid, and I'd even recommend it since it will take time for registries to support the new artifact manifest type.

Reply:
> you can use the image manifest with a custom config

Yes.  Thats exactly what I am doing now, but I was hoping that when the artifact spec is released, that I would not be having to have a technically non-standard image.  My application does not use layers so the image spec with `layers` is not technically valid.

I understand that the spec will take some time to be supported but I would like that once the time comes, I am able to use the OCI distribution without breaking the `rules`:

Here is the spec description for `layers`
> The array MUST have the base layer at index 0. Subsequent layers MUST then follow in stack order (i.e. from layers[0] to layers[len(layers)-1]). The final filesystem layout MUST match the result of [applying](https://github.com/opencontainers/image-spec/blob/main/layer.md#applying-changesets) the layers to an empty directory. The [ownership, mode, and other attributes](https://github.com/opencontainers/image-spec/blob/main/layer.md#file-attributes) of the initial empty directory are unspecified.

By adding another blob, it is making it more difficult for existing registry distributions to be used because (for example) harbor already reads the `config` for an image automatically do display in the UI.  So now anyone that wants to use an artifact will need to make their own custom flows that are specific to their own use case to download an individual arbitrary blob from the image as opposed to just downloading a standard manifest config.

Your point is valid that these can be worked around, but I am hoping that the solution is built flexible enough that novel use cases are not `hacks` but can be widely used across all OCI-compliant registry implementations.

Reply:
The current goal of the working group was to support attaching references and that was the primary motivation of creating the artifact manifest for the reference type working group without having to hack the image manifest to attach blobs of signatures or sboms. 

The artifact manifest does have potential, and it would be nice to consider if we want to expand its scope to help define other types of images/applications. If we do want to enhance the scope of the artifact manifest I would recommend we can do this once reference type work is merged and discuss adding more fields (like the data field in the descriptor https://github.com/opencontainers/image-spec/pull/826). 

Does that sound reasonable?

Reply:
> By adding another blob, it is making it more difficult for existing registry distributions to be used because (for example) harbor already reads the `config` for an image automatically do display in the UI. So now anyone that wants to use an artifact will need to make their own custom flows that are specific to their own use case to download an individual arbitrary blob from the image as opposed to just downloading a standard manifest config.

I haven't pulled up the Harbor UI in a while. What metadata are they extracting from the `config`? One of the goals we had was to move the config mediaType to artifactType, and then any data could be an annotation. I'm wondering if we have all the data needed without introducing another blob.

There's also nothing preventing us from defining a new artifact metadata json schema in the future, with a well known media type, that could be added as a blob to any artifact.

Reply:
Thanks for the feedback.  I am certainly not wanting to derail any efforts.  I will wait for the spec to get merged and then will post any feedback as an issue and we can address in the future.

> What metadata are they extracting from the config?

Harbor reads the JSON in the config file and displays it directly in the UI:

config.json
```
{
    "flows": [
        "flow.wafl"
    ],
    "permissions": [
        "permissions.wafl"
    ],
    "interfaces": [
        "add.apex",
        "delete.apex",
        "edit.apex",
        "list.apex"
    ]
}
```
<img width="965" alt="image" src="https://user-images.githubusercontent.com/99134191/185533260-9ab64d7d-d0f3-4667-860e-1cf52432c981.png">